### ISSUE TITLE
[feature] Components: POC components in the cmake generator

### ISSUE THREAD

**Issue by danimtb** [conan-io/conan#6718]
After #6716 and #6717, we should be able to do a POC with the CMake generator and the usage of the "smart targets": Targets that depend only on the components needed (not the whole library. 

With the possibility of linking just with the needed components of other requirements depending on the implementation of #6717

**Comment by syoliver** [comment]
I have a very poc for cmake/cmake_find_package, and some questions on this feature [here](https://github.com/syoliver/conan/tree/add-components-to-generators). How can I help @danimtb ?

**Comment by danimtb** [comment]
Hi @syoliver and thanks a lot for pointing me to your branch!

In fact, there are some good ideas there.

I have opened a new PR #7108 with my implementation using a jinja2 template. It will be great if you could review it or even try my branch from your side 😄

**Comment by syoliver** [comment]
@danimtb Your PR #7108 seems to work for the cmake_find_package generator part of my recipe :+1:

**Comment by danimtb** [comment]
@syoliver thanks a lot for testing!! We are in the review process right now and we will hopefully have components experimental implementation for the `cmake_find_package`` generator in the next Conan release!

**Comment by syoliver** [comment]
> @syoliver thanks a lot for testing!! We are in the review process right now and we will hopefully have components experimental implementation for the `cmake_find_package`` generator in the next Conan release!

Will you work on cmake generator after? If I can help you...

**Comment by danimtb** [comment]
We have to consider the implementation for the `cmake` generator. Currently we are pushing towards promoting generators such as `cmake_find_package/_multi` generators as we think the interface using `find_package()` is cleaner.

Moreover, we are implementing a new "toolchain" concept #7076 that will cover the functionality of macros such as `conan_basic_setup()`.

Anyway, as I see you are interested we can ping you in the PRs related to components and CMake so we can discuss it.

**Comment by memsharded** [comment]
#7108 only provided ``cmake_find_package`` support, not ``cmake``. Reopening this.
